k9mail/src/main/java/com/fsck/k9/helper/Utility.java

```diff
@@ -459,6 +472,45 @@ public static String extractMessageId(final String text) {
         return null;
     }
 
+    public static String getResizedImageFile(Context context, Uri uri, float multiplier){
+        File cacheDir = context.getCacheDir();
+        File tempAttachmentsDirectory = new File(cacheDir.getPath() + "/tempAttachments/");
```

Comment:
Did you look at using `TemporaryAttachmentStore` ? This seems to work very similarly.